workers/generic-worker/graceful/graceful.go

```diff
@@ -0,0 +1,53 @@
+package graceful
+
+import "sync"
+
+type GracefulTerminationFunc func(bool)
+
+var (
+	// Mutex for access to other vars
+	m sync.Mutex
+
+	// True if a graceful termination has been requestd
+	terminationRequested bool
+
+	// pending callback for graceful termination (there can be only one)
+	callback GracefulTerminationFunc
+)
+
+// Return true if graceful termination has been requested
+func TerminationRequested() bool {
+	return terminationRequested
```

Comment:
We'll need to lock the mutex for reads too. Note, you could use a [`sync.RWMutex`](https://golang.org/pkg/sync/#RWMutex) for marginal gain (reads won't block other reads), although the `sync.Mutex` is also fine.

Concurrency is tricky.